project/src/test/resources/docker/sql/create-schema.sql

```diff
@@ -469,3 +475,223 @@ CREATE TABLE dbo.policies(
 	version VARCHAR(10),	--	v0, v1...
 	path VARCHAR(100)		-- .../courses
 );
+
+-----------------------------------------------------
+-- Write API
+
+CREATE OR REPLACE PROCEDURE dbo.sp_createOrReplaceSchool (
+  schoolName VARCHAR(100),
+  schoolAcr VARCHAR(50),
+  programmeName VARCHAR(100),
+  programmeAcr VARCHAR(50),
+  programmeTermSize INT,
+  calendarTerm VARCHAR(50))
+AS $$
+#print_strict_params ON
+BEGIN
+  IF (schoolName IS NULL AND schoolAcr IS NULL) OR (programmeAcr IS NULL AND programmeName IS NULL) THEN
+    RAISE 'For the School and Programme parameters, you must provide at least a "name" or an "acronym"';
+  END IF;
+
+  -- Insert or update Programme
+  -- coalesce: if any of the supplied parameters is null, use the previous value of the column instead
+  UPDATE
+    dbo.programme
+  SET
+    name     = COALESCE(programmeName, name),
+    acronym  = COALESCE(programmeAcr, acronym),
+    termSize = COALESCE(programmeTermSize, termSize)
+  WHERE
+    name = programmeName OR acronym = programmeAcr;
+
+  INSERT INTO
+    dbo.Programme(name, acronym, termSize)
+  SELECT
+    programmeName, programmeAcr, programmeTermSize
+  WHERE
+    NOT EXISTS(
+      SELECT id FROM dbo.Programme p
+      WHERE p.name = programmeName OR p.acronym=programmeAcr
+    );
```

Comment:
I think this could be made more clear with an `INSERT ON CONFLICT` clause.
```sql
INSERT INTO
  dbo.programme(name, acronym, termSize)
VALUES
 (
    COALESCE(programmeName, name),
    COALESCE(programmeAcr, acronym),
    COALESCE(programmeTermSize, termSize)
 )
ON CONFLICT DO UPDATE 
  SET
    name     = COALESCE(programmeName, name),
    acronym  = COALESCE(programmeAcr, acronym),
    termSize = COALESCE(programmeTermSize, termSize);
``` 

I'm not sure the statement would be exactly like this but it's something close.